ui/app/common/displaycontrols/forms/controllers/versionedFormController.js

```diff
@@ -40,30 +44,71 @@ angular.module('bahmni.common.displaycontrol.forms')
-                                        formName: latestForm.name,
-                                        formNameTranslations: latestForm.nameTranslation ? JSON.parse(latestForm.nameTranslation) : []
-                                    };
+                var privileges = [];
+                return $q.all([formService.getAllPatientForms($scope.patient.uuid, $scope.section.dashboardConfig.maximumNoOfVisits, $state.params.enrollment), latestPublishedForms()]).then(function (results) {
+                    if (!(results[0] && results[0].data.length)) {
+                        $scope.formsNotFound = true;
+                        $scope.$emit("no-data-present-event");
+                    } else {
+                        var formListFromObsTab = results[0] && results[0].data;
+                        var attachedFormList = [];
+                        var latestForms = results[1] && results[1].data;
+                        var privileges = 'privileges';
+
+                        if (latestForms) {
+                            $scope.formsWithNameTranslations = latestForms.map(function (latestForm) {
+                                _.each(formListFromObsTab, function (item) {
+                                    if (item.formName === latestForm.name) {
+                                        item['privileges'] = latestForm.privileges;
+                                        attachedFormList.push(item);
+                                    }
                                 });
-                            }
-                            var sortedFormDataByDate = sortFormDataByLatestDate(filterForms(results[0].data));
-                            if ($scope.isOnDashboard) {
-                                $scope.formData = getUniqueForms(sortedFormDataByDate);
-                            } else {
-                                $scope.formData = sortedFormDataByDate;
-                            }
+                                return {
+                                    formName: latestForm.name,
+                                    formNameTranslations: latestForm.nameTranslation ? JSON.parse(latestForm.nameTranslation) : []
+
+                                };
+                            });
+                        }
+                        if (attachedFormList.length == 0) {
+                            attachedFormList = formListFromObsTab;
+                        }
+                        var sortedFormDataByDate = sortFormDataByLatestDate(filterForms(attachedFormList));
+                        if ($scope.isOnDashboard) {
+                            $scope.formData = getUniqueForms(sortedFormDataByDate);
+                        } else {
+                            $scope.formData = sortedFormDataByDate;
                         }
+                    }
+                });
+            };
+            $scope.doesUserHaveAccessToTheForm = function (data, action) {
+                if (data.privileges.length > 0) {
```

Comment:
@buvaneswari-arun please add null check for data.privileges(same as above)

Reply:
Done